src/TiledArray/math/linalg/rank-local.cpp

```diff
@@ -22,201 +22,187 @@
  *
  */
 
-#include <TiledArray/math/linalg/rank-local.h>
+#include <TiledArray/math/blas.h>
 #include <TiledArray/math/lapack.h>
+#include <TiledArray/math/linalg/rank-local.h>
 
-template<class F, typename ... Args>
-inline int ta_lapack_call(F f, Args ... args) {
+template <class F, typename... Args>
+inline int ta_lapack_fortran_call(F f, Args... args) {
   lapack_int info;
-  auto ptr = [](auto &&a) {
+  auto ptr = [](auto&& a) {
     using T = std::remove_reference_t<decltype(a)>;
-    if constexpr (std::is_pointer_v<T>) return a;
-    else return &a;
+    if constexpr (std::is_pointer_v<T>)
+      return a;
+    else
+      return &a;
   };
   f(ptr(args)..., &info);
   return info;
 }
 
 #define TA_LAPACK_ERROR(F) throw std::runtime_error("lapack::" #F " failed")
 
-#define TA_LAPACK_CALL(F, ARGS...) \
-  ((ta_lapack_call(F##_, ARGS) == 0) || (TA_LAPACK_ERROR(F), 0))
-
-/// TA_LAPACK(fn,args) can be called only from template context, with `T`
-/// defining the element type
-#define TA_LAPACK(name, args...) {                                             \
-  using numeric_type = T;                                                      \
-  if constexpr (std::is_same_v<numeric_type, double>)                          \
-                 TA_LAPACK_CALL(d##name, args);                                \
-  else if constexpr (std::is_same_v<numeric_type, float>)                      \
-                 TA_LAPACK_CALL(s##name, args);                                \
-  else std::abort();                                                           \
+#define TA_LAPACK_FORTRAN_CALL(F, ARGS...) \
+  ((ta_lapack_fortran_call(F, ARGS) == 0) || (TA_LAPACK_ERROR(F), 0))
+
+/// \brief Invokes the Fortran LAPACK API
+
+/// \warning TA_LAPACK_FORTRAN(fn,args) can be called only from template
+/// context, with `T` defining the element type
+#define TA_LAPACK_FORTRAN(name, args...)                    \
+  {                                                         \
+    using numeric_type = T;                                 \
+    if constexpr (std::is_same_v<numeric_type, double>)     \
+      TA_LAPACK_FORTRAN_CALL(d##name, args);                \
+    else if constexpr (std::is_same_v<numeric_type, float>) \
+      TA_LAPACK_FORTRAN_CALL(s##name, args);                \
+    else                                                    \
+      std::abort();                                         \
   }
 
+/// TA_LAPACK(fn,args) invoked lapack::fn directly and checks the return value
+#define TA_LAPACK(name, args...) \
+  ((::lapack::name(args) == 0) || (TA_LAPACK_ERROR(name), 0))
+
 namespace TiledArray::math::linalg::rank_local {
 
+using integer = math::blas::integer;
+
 template <typename T>
 void cholesky(Matrix<T>& A) {
-  char uplo = 'L';
-  lapack_int n = A.rows();
+  auto uplo = lapack::Uplo::Lower;
+  integer n = A.rows();
   auto* a = A.data();
-  lapack_int lda = n;
+  integer lda = n;
   TA_LAPACK(potrf, uplo, n, a, lda);
```

Comment:
Shouldn't we just be using `lapack::potrf` here?

Reply:
`TA_LAPACK(potrf,...` resolves to that, but also checks the return and if needed throws exception with embedded `__FILE__` and `__LINE__`